### DISCUSSION TITLE
Classic Customization in Playlist loses partial toggles

### DISCUSSION THREAD

**Discussion by itotallygetthat** [ppy/osu#16968] [Q&A] [answered]
[network.log](https://github.com/ppy/osu/files/8121902/network.log)
[runtime.log](https://github.com/ppy/osu/files/8121903/runtime.log)
[performance.log](https://github.com/ppy/osu/files/8121904/performance.log)
[updater.log](https://github.com/ppy/osu/files/8121905/updater.log)
[database.log](https://github.com/ppy/osu/files/8121906/database.log)

Lazer 2022.217.0

Game is giving an error when attempting to start a map which is in a playlist if this condition is met:
- map was added to the playlist with the Classic options enabled, customized so that only some of the Classic options are toggled on.

The error log mentions "Current Mods do not match PlaylistItem's RequiredMods".

As of a week ago (version 2022.205.0), when some Classic options were toggled, the hover text would say "XYZ option set to False," for each Classic option toggled to off.  Today (version 2022.217.0) the same hover text just says "Classic".

A map with full Classic options starts ok.
A map without Classic enabled starts ok.
Partial Classic not in a playlist starts ok.
See the playlist "partial classic test" for an example.  (https://osu.ppy.sh/multiplayer/rooms/150654)

My guess is that the client is no longer getting the Classic customization options loaded in properly in the playlist map selection menu from the server's playlist, but the server's playlist has the correct Classic customization options stored properly.
And that something since the 205 update led to this behavior.

**Comment by peppy** [comment]
Thanks for the detailed report, I'm going to look into this.

**Reply by peppy** [reply]
Apparently i am not going to look into this..?

Investigation so far: it seems to work fine on the dev server (server is returning correct values for customisation with only some settings toggled).

**Comment by smoogipoo** [comment, marked as answer]
https://github.com/ppy/osu/issues/16978
